### PR TITLE
Do not expand GetElementPtr in LLPC

### PR DESCRIPTION
**Pull request by jpages**
The llpc-spirv-lower-memory-op pass can sometimes expand
getElementPtr to a large number of instructions.
Disable this expansion here and let the backend handle
this transformation.
This will allow a better codegen in the LLVM backend.
Some cleanup in the pass as well.

See https://reviews.llvm.org/D126389 for the backend changes.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Review by piotrAMD — COMMENTED** [review]
Can we avoid expanding a dynamic index in this pass completely, and rely on the backend to handle this?

**Comment by jpages** [comment]
Thanks for the comment!

I rebased the patch. I completely removed the visit of the GetElementPtr instruction, the backend will handle this transformation.

**Comment by piotrAMD** [comment]
Thanks - looks good to me modulo the tests that need updating to match the new IR.

**Comment by jpages** [comment]
Thanks for the comments. I updated the tests and I did some additional cleanup that Jay suggested.

**Comment by jayfoad** [comment]
Can you remove m_storeExpandInfo and expandStoreInst too?

File: llpc/lower/llpcSpirvLowerMemoryOp.h [hunk 1]
```diff
@@ -61,22 +53,13 @@ class SpirvLowerMemoryOp : public SpirvLower,
...
 
 private:
-  bool needExpandDynamicIndex(llvm::GetElementPtrInst *getElemPtr, unsigned *operandIndex,
-                              unsigned *dynIndexBound) const;
-  void expandLoadInst(llvm::LoadInst *loadInst, llvm::ArrayRef<llvm::GetElementPtrInst *> getElemPtrs,
-                      llvm::Value *dynIndex);
-  void recordStoreExpandInfo(llvm::StoreInst *storeInst, llvm::ArrayRef<llvm::GetElementPtrInst *> getElemPtrs,
-                             llvm::Value *dynIndex);
   void expandStoreInst(llvm::StoreInst *storeInst, llvm::ArrayRef<llvm::GetElementPtrInst *> getElemPtrs,
                        llvm::Value *dynIndex);
```
Comment (jayfoad) [inline llpc/lower/llpcSpirvLowerMemoryOp.h:60]: Remove this too

**Comment by jpages** [comment]
Removed m_storeExpandInfo/expandStoreInst.

We are one step closer to completely remove this pass.

**Review by jayfoad — APPROVED** [review]

**Comment by piotrAMD** [comment]
> We are one step closer to completely remove this pass.

The remaining transformation can possibly be removed too, but it needs testing and is outside of the scope of this PR.

**Review by piotrAMD — APPROVED** [review]

**Review by amdrexu — DISMISSED** [review]

**Comment by Flakebi** [comment]
ok to test

**Comment by jpages** [comment]
I think Jenkins is a bit broken right now....

When I tested in local the tests were passing.

**Comment by jpages** [comment]
Rebased

**Comment by amdvlk-admin** [comment]
### Test summary for commit 39db5fe688d75fc66745f90d8413cef55df8b7ce
<details><summary>CTS tests (Failed: 0/186371)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36631/65208 (56.2%)</li>
<li>Failed: 0/65208 (0.0%)</li>
<li>Not Supported: 28577/65208 (43.8%)</li><li>Warnings: 0/65208 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29989/55953 (53.6%)</li>
<li>Failed: 0/55953 (0.0%)</li>
<li>Not Supported: 25964/55953 (46.4%)</li><li>Warnings: 0/55953 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38930/65210 (59.7%)</li>
<li>Failed: 0/65210 (0.0%)</li>
<li>Not Supported: 26280/65210 (40.3%)</li><li>Warnings: 0/65210 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**